doc/rest_api/snippet_sharing/api-description.apib

```diff
@@ -0,0 +1,410 @@
+FORMAT: 1.0
+
+# REST service API for sharing configuration snippets
+
+This is a description of the API provided by the REST service that can be used to share, search and download configuration snippets.
+
+
+# API Description [/]
+
+## List API methods [GET]
+
+Returns an overview of available API methods with required parameters and a description.
+
++ Response 200 (text/html; charset=utf-8)
+
+
+
+# Authentication endpoint [/auth]
+
+## Authenticate [POST]
+
+The API will try to authenticate the user by the submitted credentials. In case of success a session token (JWT) will be returned, which can be used for further requests to the API.
```

Comment:
For crypto it is required to compile CppCMS with more dependencies (e.g. OpenSSL), so it would only be a shift of dependencies in my eyes. There is also no documentation on how to use it anywhere (yes I looked this up before).

The main problem I have with getting rid of the Cryptlite lib is that using cryptographic functions requires lots of initialization and stuff that is done by this library. So I would basically have to write it on my own again, achieving nothing else than work. Why should I do that? License problems of other people are not my business, to be honest.